tooling/minimize/src/rvalue.rs

```diff
@@ -5,32 +5,39 @@ impl<'cx, 'tcx> FnCtxt<'cx, 'tcx> {
     pub fn translate_rvalue(
         &mut self,
         rv: &rs::Rvalue<'tcx>,
+    ) -> Option<(Vec<Statement>, ValueExpr)> {
+        self.translate_rvalue_smir(&smir::stable(rv))
+    }
+
+    pub fn translate_rvalue_smir(
+        &mut self,
+        rv: &smir::Rvalue,
     ) -> Option<(Vec<Statement>, ValueExpr)> {
         Some((
             vec![],
             match rv {
-                rs::Rvalue::Use(operand) => self.translate_operand(operand),
-                rs::Rvalue::CheckedBinaryOp(bin_op, box (l, r))
-                | rs::Rvalue::BinaryOp(bin_op, box (l, r)) => {
-                    let lty = l.ty(&self.body, self.tcx);
-                    let rty = r.ty(&self.body, self.tcx);
+                smir::Rvalue::Use(operand) => self.translate_operand_smir(operand),
+                smir::Rvalue::CheckedBinaryOp(bin_op, l, r)
+                | smir::Rvalue::BinaryOp(bin_op, l, r) => {
+                    let lty = l.ty(&self.locals_smir).unwrap();
+                    let rty = r.ty(&self.locals_smir).unwrap();
```

Comment:
When can these fail?

Reply:
This is overly protective code if the locals passed are not compatible with the projection.